### ISSUE TITLE
Proposal: New keyword `seqblk` instead of special syntax for labeled blocks.

### ISSUE THREAD

**Issue by user00e00** [ziglang/zig#4412]
Alternative to the current syntax for labeled scopes, that would be beneficial if combined with #4285 (AConf literals).

Below are examples from the documentation written with the alternative `seqblk` based syntax:

```zig
test "labeled break from labeled block expression" {
   var y: i32 = 123;
    const x = seqblk blk {
	y += 1;
	break blk, y; // label and value arguments always present
    };
	// ...
}

test "nested break" {
    while (true) seqblk outer {
        while (true) {
            break outer, _ ; // allow `_` when no value is returned from the block
        }
    }
	// ...
}

test "nested continue" {
    var i: usize = 0;
    while (i < 10) : (i += 1) seqblk outer {
        while (true) {
            continue outer;
            // `continue _ ;` to continue from innermost loop
        }
    }
	// ...
}

test "while else" {
    assert(rangeHasNumber(0, 10, 5));
    assert(!rangeHasNumber(0, 10, 15));
}

fn rangeHasNumber(begin: usize, end: usize, number: usize) bool {
    var i = begin;
    return while (i < end) : (i += 1) {
        if (i == number) {
            break _, true; // `_` as first arg means break from innermost scope
        }
    } else false;
}

const success = seqblk x {
    var i: usize = 0;
    while (i < n) : (i += 1) {
        if (found(i))
            break x, true;
    }
    break x, false;
};
```

In combination with #4285 (AConf literals), it becomes easy to add configuration options to sequential blocks. For example `.{.constantTime = true}` to take a use case from #1776 .

In short: 
```
< seqblk <blkname>? <<AConf literal> : >? > { <statement>* }
```
where `<AConf literal>` would be of the form `.{ <.config=.ConfigOption>* }`

The `seqblk` keyword would only be used when you need either a label and/or a different configuration for a scope, so in most cases code wouldn't need changing with this proposal.

```zig
seqblk blk .{.runtimeSafety = .OFF, .blocking = false} : {
// ... statements here
}

while(cond) seqblk outer { // only use label, default config
    while(cond2) seqblk .{.runtimeSafety =.OFF} : { // no label, change config
        // ...
        if(somecond){
            continue outer;
        }
    }
}

```

**Comment by MasterQ32** [comment]
I really like the approach of having a `break`-Operator that takes one operand for "break loop" and two operands for "return value from block".

But i would remove the option for "implicit block break" with `break _, …` as it's even hard to recognize what will be broken in your code examples above.

**Comment by r4pr0n** [comment]
I kinda like this proposal, but on the other hand I think the keyword seqblk would make it less readable. I agree that simplicity is important, but I think prioritises reading code over writing code and even though this change would (maybe) simplify writing the code, I think that the drawbacks in readability are more significant here.

**Comment by user00e00** [comment]
I see this issue was closed, but I will leave some more comments in case this proposal is brought up again, perhaps in relation to #4285, or #4294.

New `break` syntax, including `=>` tokens. I think this would make it more obvious when something is returned vs when a block is simply exited.
```zig
break; // normal
break => value; // when returning a value from the block
break label; // when breaking a specified label
break label => value; // when doing both of the above in 
```

This would leave the `blk: {` part unchanged, while the (to me) awkward `:blk` is done away with.
